docs/concepts/reftypes.mdx

```diff
@@ -0,0 +1,221 @@
+
+```mermaid
+sequenceDiagram
+    participant C as Client
+    participant R as Registry
+    C->>R: GET /v2/<name>/referrers/<digest>
+    R-->>C: 200 OK (List of all referrers)
+```
+
+### Listing referrers with a specific artifact type
+
+In this sequence, the client requests a list of referrers to a specific artifact
+of a certain media type from the registry, which then responds with the filtered list.
+The response will include the `OCI-Filters-Applied: artifactType` header to indicate
+that the response is filtered by the `artifactType` query parameter.
+
+The artifact type is determied by the same rules as per 
+[OCI artifacts guidlines](artifact.mdx#determining-the-artifact-type).
+
+```mermaid
+sequenceDiagram
+    participant C as Client
+    participant R as Registry
+    C->>R: GET /v2/<name>/referrers/<digest>?artifactType=<artifactType>
+    R-->>C: 200 OK (List of referrers of specific media type)
+```
+
+The request below lists shows an example for listing referrers of a specific media type.
+
+```json
+GET /v2/hello-world/referrers/sha256:5e140a6..?artifactType=application%2Fexample HTTP/1.1
+
+// response
+Content-Type: application/vnd.oci.image.index.v1+json
+Oci-Filters-Applied: artifactType
+
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.index.v1+json",
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:a10d2cf503458e48996a3e3030c75fb5f5cd967e21a38ca69ce6c1e1dee5fd27",
+      "size": 677,
+      "annotations": {
+        "org.opencontainers.image.created": "2023-08-05T07:08:43Z"
+      },
+      "artifactType": "application/example"
+    }
+  ]
+}
+```
+
+## Fetching the reference artifact
+
+Once the referrers have been queried, the client may fetch the reference artifact
+using the digest that is returned in the list of referrers.
+The client may then use the digest to fetch the artifact contents using the
+[pull workflow][pull-workflow] as described in the distribution specification.
+
+```mermaid
+sequenceDiagram
+    participant C as Client
+    participant R as Registry
+
+    Note over C,R: Pull Workflow Starts
+
+    C->>R: GET /v2/<name>/manifests/<digest>
+    R-->>C: 200 OK, return manifest
+
+    Note over C,R: If manifest has config
+
+    C->>R: GET /v2/<name>/blobs/<digest> (Config)
+    R-->>C: 200 OK, return config
+
+    Note over C,R: If manifest has blobs
+
+    C->>R: GET /v2/<name>/blobs/<digest> (Blob 1)
+    R-->>C: 200 OK/307 Redirect, return blob 1
+
+    C->>R: GET /v2/<name>/blobs/<digest> (Blob 2)
+    R-->>C: 200 OK/307 Redirect return blob 2
+
+    Note over C,R: Save to OCI Layout or file system
+
+    C->>+C: Save manifest, config and blobs 
+    Note over C: Pull Workflow Ends
+```
+
+## The Power of Associations
+
+The ability to create and list referrers opens a world of possibilities in
+managing and tracking OCI Artifacts.
+From versioning to updates, and from dependencies to signatures, the use of
```

Comment:
The design of Reference Types (Association) also empowers the following scenarios: 

- Supply chain artifacts discovery and distribution
- Artifact portability, users can easily copy a graph of OCI content across registries
- Maintain artifact lifecycle graph. For example, delete the root artifact, the associated signatures, SBOMs can all be deleted as well